python/pyarrow/tests/test_fs.py

```diff
@@ -699,12 +699,12 @@ def test_get_file_info_with_selector(fs, pathfn):
 
         infos = fs.get_file_info(selector)
         if fs.type_name == "py::fsspec+s3":
-            # s3fs only lists directories if they are not empty, but depending
-            # on the s3fs/fsspec version combo, it includes the base_dir
+            # s3fs only lists directories if they are not empty
             # (https://github.com/dask/s3fs/issues/393)
-            assert (len(infos) == 4) or (len(infos) == 5)
+            len(infos) == 4
         else:
             assert len(infos) == 5
+        assert base_dir not in infos
```

Comment:
I am not sure this would ever give false (because `infos` is a list of FileInfo objects, and `base_dir` is a path. So you would have to do something like `base_dir not in [info.path for info in infos]`, but in hindsight that's actually ocvered already by the loop below (by removing the check for the base "selector-dir", we are now actually already ensuring that it is not in the list of `infos`.

Reply:
You are correct. Will remove this lines.
(I am having trouble testing it locally as the test is failing on current master for me. Can't figure out why)